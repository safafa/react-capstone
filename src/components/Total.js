import React from 'react';
import PropTypes from 'prop-types';
import { Card } from 'react-bootstrap';

const Total = ({ name, total }) => (
  <Card className="text-center pt-3 bg-pink text-light border-0 my-1">
    <Card.Title className="title-f">{name}</Card.Title>
    <Card.Body>
      <Card.Text>
        {total}
      </Card.Text>
    </Card.Body>
  </Card>
);

Total.propTypes = {
  name: PropTypes.string,
  total: PropTypes.string,
};
Total.defaultProps = {
  name: 'name',
  total: '0',
};

export default Total;
