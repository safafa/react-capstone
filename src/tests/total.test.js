import renderer from 'react-test-renderer';
import Total from '../components/Total';

describe('Total component', () => {
  test('Snapshot test', () => {
    const tree = renderer.create(
      <Total name="World" total="12234" />,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
