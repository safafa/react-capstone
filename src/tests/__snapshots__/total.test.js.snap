// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Total component Snapshot test 1`] = `
<div
  className="text-center pt-3 bg-pink text-light border-0 my-1 card"
>
  <div
    className="title-f card-title h5"
  >
    World
  </div>
  <div
    className="card-body"
  >
    <p
      className="card-text"
    >
      12234
    </p>
  </div>
</div>
`;
